code/string_calculator/src/string_calculator.py

```diff
@@ -0,0 +1,66 @@
+import regex as re
+
+
+class StrCalculatorType(object):
+    # Data structure for storing the parsed string
+    # Abstraction for StrCalculator
+    # Let's get rid of a difficult input string,
+    # we will keep only the important details.
+    def __init__(self, string, delimiters, numstring, numbers):
+        self.string = string
+        self.delimiters = delimiters
+        self.numstring = numstring
+        self.numbers = numbers
+
+
+class StrCalculatorParser(object):
+    # string to StrCalculatorType converter
+    DEFAULT_DELIMS = ["\n", ","]
+    CUSTOM_DELIM_REGEX = "^([^\d])\n"
```

Comment:
If the regexp is constant, why not precompile it before usage? Like `re.compile(...)`?

Reply:
Done.